### ISSUE TITLE
Add Soroban interface version to getNetwork response

### ISSUE THREAD

**Issue by wildework** [stellar/stellar-rpc#17]
https://github.com/stellar/soroban-tools/blob/967351b064bac431f0d6f9ae61ba6e4f0b417a06/cmd/soroban-rpc/internal/methods/get_network.go#L31C26-L31C26

It would be very useful for the RPC method `getNetwork` to return the Soroban contract interface version alongside everything else it already returns.

Being able to fetch what Soroban version is deployed on the network exposed through the RPC would allow for a simple check before trying to deploy a new WASM. Makes the endpoint more useful.

**Comment by leighmcculloch** [comment]
@wildework made a similar suggestion that fits well with this one to have `getNetwork` start reporting what versions of the protocol are supported and that are actively in use. I think we can tie that into this issue.

> I would be super happy to see the RPC method getNetwork display the range of supported protocol and pre-protocol versions!

Ref: https://github.com/stellar/stellar-protocol/discussions/1476#discussioncomment-9035875

Similar to how Horizon already reports the versions available at:

```json
  "current_protocol_version": 20,
  "supported_protocol_version": 20,
  "core_supported_protocol_version": 20
```

Ref: https://horizon.stellar.org/
Ref: https://horizon-testnet.stellar.org/
Ref: https://horizon-futurenet.stellar.org/

**Comment by leighmcculloch** [comment]
Reflecting on this issue, it's a bit strange that the `getNetwork` method would return the `core_supported_protocol_version`, because that information has nothing to do with the network. It's information about the version of a dependency. So arguably belongs in the `getVersionInfo` method instead alongside the existing core information. 

This is what the two endpoints return today, and the information about core probably belongs in the version info method.

`getVersionInfo`:
```json
{
    "id": "1",
    "jsonrpc": "2.0",
    "result": {
        "buildTimestamp": "2025-10-20T22:54:56",
        "captiveCoreVersion": "stellar-core 24.0.0 (0d7b4345de396ad4e8d7dcc4460ddc6feeb27b11)",
        "commitHash": "f6a584ab8a409fb43f81a60dbef7854fdb06231e",
        "protocolVersion": 24,
        "version": "24.0.0-f6a584ab8a409fb43f81a60dbef7854fdb06231e"
    }
}
```

`getNetwork`:
```json
{
    "id": "1",
    "jsonrpc": "2.0",
    "result": {
        "friendbotUrl": "https://friendbot.stellar.org/",
        "passphrase": "Test SDF Network ; September 2015",
        "protocolVersion": 24
    }
}
```

**Comment by leighmcculloch** [comment]
Also, https://github.com/stellar/stellar-rpc/pull/563 makes the point that much of this information is duplicate. It's worth pointing out that the max supported protocol is always the same version as the captive core version, which is already exposed on the `getVersionInfo` endpoint. So I'm actually not sure what the use case of adding these fields are today.

Note that the original issue to report the interface version is now redundant, there is no such thing as the interface version anymore. The interface version is just the protocol version now and it is already reported.

**Comment by cjonas9** [comment]
> Also, [#563](https://github.com/stellar/stellar-rpc/pull/563) makes the point that much of this information is duplicate. It's worth pointing out that the max supported protocol is always the same version as the captive core version, which is already exposed on the `getVersionInfo` endpoint. So I'm actually not sure what the use case of adding these fields are today.
> 
> Note that the original issue to report the interface version is now redundant, there is no such thing as the interface version anymore. The interface version is just the protocol version now and it is already reported.

Great points, there are even "redundancy assurance tests" in my PR that are somewhat `if true == true` -- asserting that the core supported protocol version is the same as the highest supported ledger protocol version is always going to be true as core should refuse to start were that not the case (though the test made sense to me as it would be a good "freak accident" thing to check/catch).

To me, it still made sense to report on the `core supported protocol version` and min/max `supported ledger protocol version`s as if they were independent for clarity to whoever is hitting the endpoint rather than assuming the knowledge that they have this knowledge (especially after what versions exist have been changed a little over time).

**Comment by leighmcculloch** [comment]
> Stuff about the node belongs in getVersionInfo while stuff about the network belongs in getNetwork, so I think all we want is just coreSupportedProtocolVersion in getNetwork.

The supported protocol version of the stellar-core that rpc is running, is information about the node, not the network.

**Comment by Shaptic** [comment]
Ah yeah you're right actually 🤦 I misread https://github.com/stellar/stellar-rpc/issues/17#issuecomment-3634338047 originally. It is a bit semantic, but having the protocol stuff split (or duped?) across the two endpoints also feels bad. One could argue it's still nice to have in the sense that it's an indicator that a future version is coming to the network but honestly I don't even know who uses these endpoints for anything beyond friendbot + passphrase stuff and you're right that it's duped in the version of `getVersionInfo`.

I think this whole issue is a general note-to-self that when taking on old issues for onboarding purposes we should take care to ensure the context is accurate before we start doing them.
